ruby_event_store/lib/ruby_event_store/client.rb

```diff
@@ -339,12 +339,16 @@ def transform(events)
     end
 
     def enrich_events_metadata(events)
+      raise ArgumentError, "Event cannot be `nil`" if events.nil?
```

Comment:
Why here instead of at the top of `#publish` / `#append`?

Reply:
Because here is the guard coercing to array. I'm not sure if this is the best place to all this operations (probably not). Do you have any specific arguments for moving assertions to publish/append? (the only that I have in mind is the rule to keep assertions as close public methods as possible but I'm fine with breaking it here)

If that makes sense I can think about refactoring in separate PR

Reply:
`the only that I have in mind is the rule to keep assertions as close public methods as possible`

And that's the one I agree with :) Especially that we are raising an `ArgumentError` which is raised when arguments passed to the method are wrong. Any benefit of keeping these in `enrich_` methods?